core/geometry/src/geometry3d/Matrix3d.ts

```diff
@@ -869,377 +897,201 @@ export class Matrix3d implements BeJSONFunctions {
-    const cross20 = delta2.crossProduct(delta0);
-
-    const aa01 = cross01.magnitudeSquared();
-    const aa12 = cross12.magnitudeSquared();
-    const aa20 = cross20.magnitudeSquared();
-
-    const cross = cross01.clone(); // This will end up as the biggest cross product
-    const v0 = delta0.clone();  // This will end up as one of the two largest delta vectors
-    let aaMax = aa01;
-    if (aa12 > aaMax) {
-      cross.setFrom(cross12);
-      aaMax = aa12;
-      v0.setFrom(delta1);
-    }
-    if (aa20 > aaMax) {
-      cross.setFrom(cross20);
-      aaMax = aa20;
-      v0.setFrom(delta2);
-    }
-
-    if (aaMax === 0.0) {
-      // The vectors did not move.  Just accept the zero rotation, with error flag set.
-      return result;
+  /**
+   * Apply (in place) a jacobi update that zeros out this.at(i,j).
+   * @param i row index of zeroed member
+   * @param j column index of zeroed member
+   * @param k other row/column index (different from i and j)
+   * @param leftEigenVectors a matrix that its columns will be filled by eigenvectors of this Matrix3d
+   * (allocated by caller, computed and filled by this function)
+   */
+  private applyFastSymmetricJacobiUpdate(i: number, j: number, k: number, leftEigenVectors: Matrix3d): number {
+    const indexII = 4 * i;
+    const indexJJ = 4 * j;
+    const indexIJ = 3 * i + j;
+    const indexIK = 3 * i + k;
+    const indexJK = 3 * j + k;
+    const dotUU = this.coffs[indexII];
+    const dotVV = this.coffs[indexJJ];
+    const dotUV = this.coffs[indexIJ];
+    const jacobi = Angle.trigValuesToHalfAngleTrigValues(dotUU - dotVV, 2.0 * dotUV);
+    if (Math.abs(dotUV) < 1.0e-15 * (dotUU + dotVV))
+      return 0.0;
+    const c = jacobi.c;
+    const s = jacobi.s;
+    const cc = c * c;
+    const ss = s * s;
+    const sc2 = 2.0 * c * s;
+    this.coffs[indexII] = cc * dotUU + sc2 * dotUV + ss * dotVV;
+    this.coffs[indexJJ] = ss * dotUU - sc2 * dotUV + cc * dotVV;
+    this.coffs[indexIJ] = 0.0;
+    const a = this.coffs[indexIK];
+    const b = this.coffs[indexJK];
+    this.coffs[indexIK] = a * c + b * s;
+    this.coffs[indexJK] = -s * a + c * b;
+    this.coffs[3 * j + i] = 0.0;
+    this.coffs[3 * k + i] = this.coffs[indexIK];
+    this.coffs[3 * k + j] = this.coffs[indexJK];
+    leftEigenVectors.applyGivensColumnOp(i, j, c, s);
+    return Math.abs(dotUV);
+  }
+  /**
+   * Factor this (symmetrized) as a product U * lambda * UT where U is orthogonal, lambda is diagonal.
+   * The upper triangle is mirrored to lower triangle to enforce symmetry.
+   * @param leftEigenvectors a matrix that its columns will be filled by eigenvectors of this Matrix3d
+   * (allocated by caller, computed and filled by this function)
+   * @param lambda a vector that its entries will be filled by eigenvalues of this Matrix3d
+   * (allocated by caller, computed and filled by this function)
+   */
+  public fastSymmetricEigenvalues(leftEigenvectors: Matrix3d, lambda: Vector3d): boolean {
```

Comment:
Do not revew this function. It just moved from bottom of the file here but I have not changed the content yet.